src/StreamRequest.jl

```diff
@@ -57,7 +57,9 @@ function request(::Type{StreamLayer}, io::IO, request::Request, body;
                     writebody(http, request, body)
                 catch e
                     write_error = e
-                    @warn("Error in @async writebody task",
+                    @info("Error in @async writebody task. " *
+                          "Server likely closed the connection unexpectedly. " *
+                          "Only an issue if unable to read the response and this error gets re-thrown. ",
                           exception=(write_error, catch_backtrace()))
                     close(io)
```

Comment:
This change looks good to me; thanks for the contribution! While we're here in this code, one thing I've noticed in a few applications is that this `close(io)` will often fail because it tries to send a closing ACK or whatever to the server. I've considered adding a quick `try; close(io); catch e; end` to just swallow those errors since we don't really care whether the server received the close ACK or not. But then again, just swallowing errors always feel a little sketchy. Have you seen this at all or run into issues here?

Reply:
I haven't run into that issue but may only be because it's getting caught and retried in the code that calls HTTP.

Reply:
I can still make that change if that's something you want to add though.

Reply:
@quinnj Added

Reply:
Would `isopen(io) && close(io)` not be sufficient?